tests/ftqc/test_tape_prototype.py

```diff
@@ -129,18 +157,314 @@ def circ():
+
+        with pytest.raises(TypeError, match="cannot update 'operations'"):
+            _ = sequence.copy(operations=[X(1), Y(12)])
+
+    def test_changing_measurements_and_tapes_in_copy_raises_error(self):
+        """Test that trying to update both measurements and or tapes
+        when copying a QuantumScriptSequence raises an error"""
+
+        operations = [[X(0), Y(1)], [Y(0), Z(0)], [H(1), S(0)]]
+        measurements = [[], [], [qml.expval(X(0))]]
+        tapes = [
+            qml.tape.QuantumScript(ops, measurements=meas, shots=Shots(1000))
+            for ops, meas in zip(operations, measurements)
+        ]
+        sequence = QuantumScriptSequence(tapes)
+
+        with pytest.raises(RuntimeError, match="Can't update tapes and measurements"):
+            _ = sequence.copy(measurements=[qml.expval(X(0))], tapes=tapes)
+
+    def test_get_standard_wire_map(self):
+        """Test that getting the standard wire map works as expected"""
+
+        tape1 = qml.tape.QuantumScript([X(12), Y("a"), Z(0)])
+        tape2 = qml.tape.QuantumScript([H(1)], measurements=[qml.expval(Y(3))])
+
+        sequence = QuantumScriptSequence([tape1, tape2], shots=Shots(100))
+
+        # pylint: disable=protected-access
+        assert sequence._get_standard_wire_map() == {12: 0, "a": 1, 0: 2, 1: 3, 3: 4}
+
+    def test_map_to_standard_wires(self):
+        """Test that map_to_standard_wires correctly affects both the overall
+        sequence wires and the underlying tape wires"""
+
+        tape1 = qml.tape.QuantumScript([X(12), Y("a"), Z(0), S("b")])
+        tape2 = qml.tape.QuantumScript([H(1)], measurements=[qml.expval(Y("b"))])
+        sequence = QuantumScriptSequence([tape1, tape2], shots=Shots(100))
+
+        new_sequence = sequence.map_to_standard_wires()
+
+        assert new_sequence.tapes[0].operations == [X(0), Y(1), Z(2), S(3)]
+        assert new_sequence.tapes[1].operations == [H(4)]
+        assert new_sequence.measurements == new_sequence.tapes[1].measurements == [qml.expval(Y(3))]
+
+    def test_split_at_non_clifford_gates_creates_sequence(self):
+        """Test that split_at_non_clifford_gates splits the tape before
+        each non-Clifford gate and returns a QuantumScriptSequence"""
+
+        dev = qml.device("lightning.qubit", wires=3)
+
+        @qml.set_shots(1000)
+        @qml.qnode(dev)
+        def circ():
+            RotXZX(0.64, 0.33, 1.2, 0)
+            RotXZX(0.16, 0.93, 0.29, 1)
+            H(0)
+            S(1)
+            RZ(0.54, 1)
+            H(0)
+            S(1)
+            H(2)
+            RotXZX(0.82, 0.66, 0.26, 2)
+            H(2)
+            H(0)
+            S(1)
+            return qml.expval(X(0)), qml.expval(Y(2))
+
+        tape = qml.workflow.construct_tape(circ)()
+        sequence, fn = split_at_non_clifford_gates(tape)
+        sequence = fn(sequence)
+        sequence, fn = split_at_non_clifford_gates(tape)
+        sequence = fn(sequence)
```

Comment:
So each call affects the outcome here?
Or maybe I'm missing something about the replication?